syntax = "proto3";

service KeyValueStore {
    rpc Put (PutRequest) returns (PutResponse);
    rpc Get (GetRequest) returns (GetResponse);
    rpc Delete (DeleteRequest) returns (DeleteResponse);
}

message PutRequest {
    string key = 1;
    string value = 2;
}

message PutResponse {
    string status = 1;
}

message GetRequest {
    string key = 1;
}

message GetResponse {
    string value = 1;
    string status = 2;
}

message DeleteRequest {
    string key = 1;
}

message DeleteResponse {
    string status = 1;
}
